homeassistant/components/wolflink/sensor.py

```diff
@@ -0,0 +1,174 @@
+"""The Wolf SmartSet sensors."""
+import logging
+
+from wolf_smartset.models import (
+    HoursParameter,
+    ListItemParameter,
+    Parameter,
+    PercentageParameter,
+    Pressure,
+    SimpleParameter,
+    Temperature,
+)
+
+from homeassistant.components.wolflink.const import DOMAIN, STATES
+from homeassistant.const import (
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_TIMESTAMP,
+    PRESSURE_BAR,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up all entries for Wolf Platform."""
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+    parameters = hass.data[DOMAIN][config_entry.entry_id]["parameters"]
+
+    entities = []
+    for parameter in parameters:
+        if isinstance(parameter, Temperature):
+            entities.append(WolfLinkTemperature(coordinator, parameter))
+        if isinstance(parameter, Pressure):
+            entities.append(WolfLinkPressure(coordinator, parameter))
+        if isinstance(parameter, PercentageParameter):
+            entities.append(WolfLinkPercentage(coordinator, parameter))
+        if isinstance(parameter, ListItemParameter):
+            entities.append(WolfLinkState(coordinator, parameter))
+        if isinstance(parameter, HoursParameter):
+            entities.append(WolfLinkHours(coordinator, parameter))
+        if isinstance(parameter, SimpleParameter):
+            entities.append(WolfLinkSensor(coordinator, parameter))
+
+    async_add_entities(entities, True)
+
+
+class WolfLinkSensor(Entity):
+    """Base class for all Wolf entities."""
+
+    def __init__(self, coordinator, wolf_object: Parameter):
+        """Initialize."""
+        self.coordinator = coordinator
+        self.wolf_object = wolf_object
+
+    @property
+    def name(self):
+        """Return the name."""
+        return f"{self.wolf_object.name}"
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self.coordinator.data[self.wolf_object.value_id]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "parameter_id": self.wolf_object.parameter_id,
+            "value_id": self.wolf_object.value_id,
+            "parent": self.wolf_object.parent,
+        }
+
+    @property
+    def unique_id(self):
+        """Return a unique_id for this entity."""
+        return self.wolf_object.parameter_id
```

Comment:
If the user adds a second wolflink integration instance the unique_id won't be unique anymore, correct? We need to add something that is unique per user account eg the username or even better if there's some other unique id for the account.

Reply:
I did the reverse engineering for this API so I am not sure about paramter_id uniqueness but for this API is so complicated that this is very unique. Anyway I will add device_id also as a part of unique_id. I think that you are not able to have the same device_id on two different accounts.